recipes-extended/xen/blktap3/fix-run-time-errors-and-memory-leaks.patch

```diff
@@ -0,0 +1,270 @@
+ 			if (ret <= 0)
+ 				break;
+ 			offset += ret;
+@@ -120,7 +120,7 @@ tap_ctl_write_message(int fd, tapdisk_me
+ 		if (ret == -1)
+ 			break;
+ 		else if (FD_ISSET(fd, &writefds)) {
+-			ret = write(fd, message + offset, len - offset);
++			ret = write(fd, (char*)message + offset, len - offset);
+ 			if (ret <= 0)
+ 				break;
+ 			offset += ret;
+Index: blktap/include/blktap.h
+===================================================================
+--- blktap.orig/include/blktap.h
++++ blktap/include/blktap.h
+@@ -32,7 +32,7 @@
+ #define _TD_BLKTAP_H_
+ 
+ #define BLKTAP2_SYSFS_DIR              "/sys/class/blktap2"
+-#define BLKTAP2_CONTROL_NAME           "blktap/control"
++#define BLKTAP2_CONTROL_NAME           "blktap-control"
+ #define BLKTAP2_CONTROL_DIR            "/var/run/blktap-control"
+ #define BLKTAP2_CONTROL_SOCKET         "ctl"
+ #define BLKTAP2_DIRECTORY              "/dev/xen/blktap-2"
+Index: blktap/include/blktap2.h
+===================================================================
+--- blktap.orig/include/blktap2.h
++++ blktap/include/blktap2.h
+@@ -50,7 +50,7 @@
+ #define BLKTAP2_IOCTL_REMOVE_DEVICE    207
+ 
+ #define BLKTAP2_SYSFS_DIR              "/sys/class/blktap2"
+-#define BLKTAP2_CONTROL_NAME           "blktap/control"
++#define BLKTAP2_CONTROL_NAME           "blktap-control"
+ #define BLKTAP2_CONTROL_DIR            "/var/run/blktap-control"
+ #define BLKTAP2_CONTROL_SOCKET         "ctl"
+ #define BLKTAP2_DIRECTORY              "/dev/xen/blktap-2"
+Index: blktap/tapback/frontend.c
+===================================================================
+--- blktap.orig/tapback/frontend.c
++++ blktap/tapback/frontend.c
+@@ -478,7 +478,7 @@ tapback_backend_handle_otherend_watch(ba
+ {
+     vbd_t *device = NULL;
+     int err = 0, state = 0;
+-    char *s = NULL, *end = NULL, *_path = NULL;
++    char *s = NULL, *end = NULL;
+ 
+ 	ASSERT(backend);
+     ASSERT(path);
+@@ -509,30 +509,7 @@ tapback_backend_handle_otherend_watch(ba
+      */
+ 	s = tapback_xs_read(device->backend->xs, XBT_NULL, "%s",
+ 			device->frontend_state_path);
+-    if (!s) {
+-        err = errno;
+-		/*
+-         * If the front-end XenBus node is missing, the XenBus device has been
+-         * removed: remove the XenBus back-end node.
+-		 */
+-		if (err == ENOENT) {
+-            err = asprintf(&_path, "%s/%s/%d/%d", XENSTORE_BACKEND,
+-                    device->backend->name, device->domid, device->devid);
+-            if (err == -1) {
+-                err = errno;
+-                WARN(device, "failed to asprintf: %s\n", strerror(err));
+-                goto out;
+-            }
+-            err = 0;
+-            if (!xs_rm(device->backend->xs, XBT_NULL, _path)) {
+-                if (errno != ENOENT) {
+-                    err = errno;
+-                    WARN(device, "failed to remove %s: %s\n", path,
+-                            strerror(err));
+-                }
+-            }
+-		}
+-    } else {
```

Comment:
I think the code removed is dead code. Above this code there is a comment mentioning non-existence of front-end path is not possible as that is what triggers the function call.  Also, we search a device with appropriate front-end state path above. The function returns if device is NULL which could be because of non-existent front-end state path.
However, please let me know if you think this is not dead code. I'll remove this hunk from patch.